vos/vos/vos.py

```diff
@@ -761,6 +771,54 @@ def node_list(self):
                     self.add_child(nodeNode)
         return self._node_list
 
+    def get_children(self, client, sort, order, limit=500):
+        """ Gets an iterator over the nodes held to by a ContainerNode"""
+        # IF THE CALLER KNOWS THEY DON'T NEED THE CHILDREN THEY
+        # CAN SET LIMIT=0 IN THE CALL Also, if the number of nodes
+        # on the firt call was less than 500, we likely got them
+        # all during the init
+        if not self.isdir():
+            return
+
+        if self.node_list is not None:
+            # children already downloaded
+            for i in self.node_list:
+                yield i.uri
+
+        # stream children
+        xml_file = StringIO(
+            client.open(self.uri, os.O_RDONLY,
+                        limit=limit, sort=sort,
+                        order=order).read().decode('UTF-8'))
+        xml_file.seek(0)
+        page = Node(ElementTree.parse(xml_file).getroot())
+        nl = page.node_list
+        current_index = 0
+        run = True
+        while run and nl:
+            yield_node = nl[current_index]
+            current_index = current_index + 1
+            if current_index == len(nl):
+                if len(nl) == limit:
+                    # do another page read
+                    xml_file = StringIO(
+                        client.open(self.uri, os.O_RDONLY, next_uri=nl[-1].uri,
+                                    sort=sort, order=order,
+                                    limit=limit).read().decode('UTF-8'))
+                    xml_file.seek(0)
+                    page = Node(ElementTree.parse(xml_file).getroot())
+                    nl = page.node_list
+                    if len(nl) == 1 and nl[0].uri == yield_node.uri:
+                        # that must be the last node
+                        run = False
+                    else:
+                        current_index = 0
+                else:
+                    run = False
+            with client.nodeCache.watch(yield_node.uri) as childWatch:
+                childWatch.insert(yield_node)
```

Comment:
I'm not sure what this does?

Reply:
I wasn't sure either. I thought it's meant to add the file info to the local cache. This is what get_node does with its children but I can remove it if you think otherwise.